django/forms/widgets.py

```diff
@@ -373,13 +374,43 @@ class FileInput(Input):
     needs_multipart_form = True
     template_name = 'django/forms/widgets/file.html'
 
+    def __init__(self, attrs=None, multiple=None):
+        if attrs is None:
+            attrs = {}
+        if multiple is None:
+            try:
+                multiple = attrs['multiple']
+            except KeyError:
+                multiple = False
+            else:
+                warnings.warn(
+                    'The attr "multiple" is deprecated in favor of the keyword argument "multiple".',
+                    RemovedInDjango30Warning,
+                )
+        self.multiple = multiple
+        if self.multiple:
+            attrs.setdefault('multiple', True)
+        super().__init__(attrs=attrs)
```

Comment:
if I pass `attrs={'multiple': False)` and init with multiple=True, they are out of sync.

Reply:
@melvyn-sopacua that assessment is correct. If you'd to write
```python
ClearableFileInput(multiple=True, attrs={'multiple': False})
```
you would end up with a HTML input that only allows one file, but you would get a list containing 0..1 file in Django. I agree the use case for that is slim. I could raise a warning, what do you think?